### PR TITLE
Move bam_create_som_pon_gatk to nf-test

### PR DESCRIPTION
**Pull request by matthdsm**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by matthdsm** [comment]
Now we just need to actually fix the subworkflow

File: subworkflows/nf-core/bam_create_som_pon_gatk/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert workflow.out.mutect2_vcf.get(0).get(1) ==~ "*.vcf.gz" }
+                { assert workflow.out.mutect2_index.get(0).get(1) ==~ "*.vcf.gz.tbi" }
+                { assert workflow.out.mutect2_stats.get(0).get(1) ==~"*.stats"  }
+                { assert workflow.out.pon_vcf.get(0).get(1) ==~ "*.vcf.gz" }
+                { assert workflow.out.pon_index.get(0).get(1) ==~ "*.vcf.gz.tbi" }
```
Comment (nvnieuwk) [inline subworkflows/nf-core/bam_create_som_pon_gatk/tests/main.nf.test:43]: Why no snapshots here?

**Comment by matthdsm** [comment]
I'm obviously doing something wrong here.. Anyone got any remarks or hints/fixes?

File: subworkflows/nf-core/bam_create_som_pon_gatk/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert workflow.success },
+                { assert snapshot(file(process.out.mutect2_vcf.get(0).get(1)).name).match("test1.vcf.gz") },
+                { assert snapshot(file(process.out.mutect2_index.get(0).get(1)).name).match("test1.vcf.gz.tbi") },
+                { assert snapshot(file(process.out.mutect2_stats.get(0).get(1)).name).match("test1.vcf.gz.stats") },
+                { assert snapshot(file(process.out.pon_vcf.get(0).get(1)).name).match("test_panel.vcf.gz") },
+                { assert snapshot(file(process.out.pon_index.get(0).get(1)).name).match("test_panel.vcf.gz.tbi") },
```
Comment (nvnieuwk) [inline subworkflows/nf-core/bam_create_som_pon_gatk/tests/main.nf.test:43]:
```suggestion
                { assert snapshot(file(workflow.out.mutect2_vcf.get(0).get(1)).name).match("test1.vcf.gz") },
                { assert snapshot(file(workflow.out.mutect2_index.get(0).get(1)).name).match("test1.vcf.gz.tbi") },
                { assert snapshot(file(workflow.out.mutect2_stats.get(0).get(1)).name).match("test1.vcf.gz.stats") },
                { assert snapshot(file(workflow.out.pon_vcf.get(0).get(1)).name).match("test_panel.vcf.gz") },
                { assert snapshot(file(workflow.out.pon_index.get(0).get(1)).name).match("test_panel.vcf.gz.tbi") },
```

**Comment by matthdsm** [comment]
Tests are passing, thanks @nvnieuwk 
Now I just need to find a way to fix this linting issue
https://github.com/nf-core/modules/actions/runs/7887990901/job/21524466262?pr=4904

**Pull request merged**